### ISSUE TITLE
Bug when writing MIDI with overlapping notes

### ISSUE THREAD

**Issue by Natooz** [YatingMusic/miditoolkit#27]
Following #14, notes are now loaded FIFO, which is the intended process.
However it seems to have a sorting / ordering bug when writing the MIDI messages with overlapping notes, with the same onset times. In particular, the velocity values might end up mixed.
(I found the bug as it broke miditok's tests.)

To reproduce the error (download [Mr. Blue Sky.mid.zip](https://github.com/YatingMusic/miditoolkit/files/13219861/Mr.Blue.Sky.mid.zip) and unzip first):

```Python
# Loading is all good
midi = MidiFile("Mr. Blue Sky.mid")
# Writing it unchanged
midi.dump("Mr. Blue Sky 2.mid")
# Loading it back
midi2 = MidiFile("Mr. Blue Sky 2.mid")
# Sorting the notes, as after dump the order might have changed
midi.instruments[4].notes.sort(key=lambda x: (x.start, x.pitch, x.end, x.velocity))
midi2.instruments[4].notes.sort(key=lambda x: (x.start, x.pitch, x.end, x.velocity))

# The bug occurs on the Strings track
for note_1, note_2 in zip(midi.instruments[4].notes, midi2.instruments[4].notes):
    assert note_1.pitch == note_2.pitch
    assert note_1.start == note_2.start
    assert note_1.end == note_2.end
    assert note_1.velocity == note_2.velocity  # fails here for note 285
```

I'll work on it very soon.

**Comment by Natooz** [comment]
The problem in image:

<img width="551" alt="Capture d’écran 2023-11-01 à 11 30 50" src="https://github.com/YatingMusic/miditoolkit/assets/56734983/c4c4a94e-f1fe-4429-84df-9a45e2c43547">

Here for the `note_on` messages for the two notes with pitch 65, the one from the note ending first (with velocity 105) should come first to respect the FIFO logic. Here after writing the MIDI file both notes are present and with the right onset and offset absolute timings, but their velocities have been exchanged.

Working on putting the messages in the right order.

**Comment by Natooz** [comment]
Fixed in #28.
